extensions/narayana-jta/deployment/src/main/java/io/quarkus/narayana/jta/deployment/NarayanaJtaProcessor.java

```diff
@@ -204,6 +204,11 @@ public void startRecoveryService(NarayanaJtaRecorder recorder,
     @BuildStep(onlyIf = IsTest.class)
```

Comment:
Wonder if making it a predicate and using it in `onlyIf` would be better

Reply:
The reason I want to avoid that is that we haven't really specified what the TCCL is in those pieces of code